lang/test/org/partiql/lang/eval/EvaluatingCompilerOrderByTests.kt

```diff
@@ -0,0 +1,95 @@
+            { customerId: 456, sellerId: 2, productId: 44444, cost: 4 },
+            { customerId: 456, sellerId: 1, productId: 55555, cost: 5 },
+            { customerId: 456, sellerId: 2, productId: 66666, cost: 6 },
+            { customerId: 789, sellerId: 1, productId: 77777, cost: 7 },
+            { customerId: 789, sellerId: 2, productId: 88888, cost: 8 },
+            { customerId: 789, sellerId: 1, productId: 99999, cost: 9 },
+            { customerId: 100, sellerId: 2, productId: 10000, cost: 10 }
+        ]""",
+        "employees" to """[
+             { name: 'Joey', age: 25, manager: { name: 'John', address: { city: 'Seattle' } } },
+             { name: 'Chandler', age: 27, manager: { name: 'Rocky', address: { city: 'Seattle' } } },
+             { name: 'Ross', age: 22, manager: { 'name': 'Alex', address: { city: 'Chicago' } } }
+        ]"""
+    ).toSession()
+
+    class ArgsProviderValid : ArgumentsProviderBase() {
+        override fun getParameters(): List<Any> = listOf(
+            // should order by col1 asc
+            EvaluatorTestCase(
+                "SELECT col1 FROM simple_1 ORDER BY col1",
+                "<<{'col1': 1}, {'col1': 1}, {'col1': 1}, {'col1': 3}, {'col1': 5}>>"
+            ),
+            // should order by col1 desc
+            EvaluatorTestCase(
+                "SELECT col1 FROM simple_1 ORDER BY col1 DESC",
+                "<<{'col1': 5}, {'col1': 3}, {'col1': 1}, {'col1': 1}, {'col1': 1}>>"
+            ),
+            // should order by col1 and then col2 asc
+            EvaluatorTestCase(
+                "SELECT * FROM simple_1 ORDER BY col1, col2",
+                "<<{'col1': 1, 'col2': 5}, {'col1': 1, 'col2': 7}, {'col1': 1, 'col2': 10}, {'col1': 5, 'col2': 7}, {'col1': 3, 'col2': 12}>>"
+            ),
+            // should order by price desc and productId asc
+            EvaluatorTestCase(
+                "SELECT * FROM products ORDER BY price DESC, productId ASC LIMIT 1",
+                "<<{ 'productId': 3, 'supplierId': 10, 'categoryId': 21, 'price': 15.0, 'numInStock': 100 }>>"
+            ),
+            // should order by supplierId_nulls nulls last
+            EvaluatorTestCase(
+                "SELECT productId FROM products_sparse ORDER BY supplierId_nulls NULLS LAST, productId",
+                "<<{'productId': 1}, {'productId': 2}, {'productId': 3}, {'productId': 6}, {'productId': 7}, {'productId': 4}, {'productId': 5}, {'productId': 8}, {'productId': 9}, {'productId': 10}>>"
+            ),
+            // should order by supplierId_nulls nulls first
+            EvaluatorTestCase(
+                "SELECT productId FROM products_sparse ORDER BY supplierId_nulls NULLS FIRST, productId",
+                "<<{'productId': 4}, {'productId': 5}, {'productId': 8}, {'productId': 9}, {'productId': 10}, {'productId': 1}, {'productId': 2}, {'productId': 3}, {'productId': 6}, {'productId': 7}>>"
+            )
```

Comment:
nit: may be helpful to include `supplierId_nulls` in the select projection to verify the results